From 09e6186e134831d67f614266801a45a3a70f7a64 Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Mon, 7 Mar 2011 22:39:37 +0100
Subject: [PATCH] systemd: use standard syslog.socket unit

In systemd we now have a standard socket unit for /dev/log called
syslog.socket. This unit can be shared between an early boot mini syslog
and the full syslog implementation started later on. The mini syslog is
shipped along systemd and does nothing but simply forward the data
received through /dev/log to the kernel log buffer (i.e. kmsg, as
visible by dmesg). It is run during early boot, and then as soon as
rsyslog starts up it is terminated, so that rsyslog can take over the
/dev/log socket. Since one of the first things rsyslog does after
starting up is flushing the kernel log buffer to disk we end up with all
data from early boot up in syslog.

This patch changes two things: removes rsyslog.socket and instead
configures rsyslog.service to take possession of syslock.socket. And
secondly includes a PreStart line to terminate the running syslog bridge
instance.
---
 Makefile.am        |    3 ---
 rsyslog.service.in |    3 ++-
 rsyslog.socket     |    8 --------
 3 files changed, 2 insertions(+), 12 deletions(-)
 delete mode 100644 rsyslog.socket

diff --git a/Makefile.am b/Makefile.am
index b63e9ec..9916bf0 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -44,9 +44,6 @@ endif
 #
 if HAVE_SYSTEMD
 
-dist_systemdsystemunit_DATA = \
-	rsyslog.socket
-
 nodist_systemdsystemunit_DATA = \
 	rsyslog.service
 
diff --git a/rsyslog.service.in b/rsyslog.service.in
index 2bcde52..03db596 100644
--- a/rsyslog.service.in
+++ b/rsyslog.service.in
@@ -2,9 +2,10 @@
 Description=System Logging Service
 
 [Service]
+ExecStartPre=/bin/systemctl stop systemd-kmsg-syslogd.service
 ExecStart=@sbindir@/rsyslogd -n -c5
 ExecReload=/bin/kill -HUP $MAINPID
+Sockets=syslog.socket
 
 [Install]
 WantedBy=multi-user.target
-Also=rsyslog.socket
diff --git a/rsyslog.socket b/rsyslog.socket
deleted file mode 100644
index 0cd8605..0000000
--- a/rsyslog.socket
+++ /dev/null
@@ -1,8 +0,0 @@
-[Unit]
-Description=Syslog Socket
-
-[Socket]
-ListenDatagram=/dev/log
-
-[Install]
-WantedBy=sockets.target
-- 
1.7.4.1

